projects/Mallard/src/components/weather.tsx

```diff
@@ -138,79 +143,96 @@ const WeatherIconView = ({
-                                    />
-                                )
-                            })}
+            <View style={styles.weatherContainerLong}>
+                <GridRowSplit
+                    proxy={
+                        <View style={styles.locationNameContainer}>
+                            {showSetLocationButton ? (
+                                <Button
+                                    onPress={onSetLocationPress}
+                                    appearance={ButtonAppearance.skeleton}
+                                    style={styles.setLocationButtonWrap}
+                                    buttonStyles={styles.setLocationButton}
+                                >
+                                    Set Location
+                                </Button>
+                            ) : (
+                                <>
+                                    <Text style={styles.locationPinIcon}>
+                                        {'\uE01B'}
+                                    </Text>
+                                    <Text style={styles.locationName}>
+                                        {locationName}
+                                    </Text>
+                                </>
+                            )}
                         </View>
-                    ),
-                }}
-            </WithBreakpoints>
+                    }
+                >
+                    {intervals.map(forecast => {
+                        return (
+                            <WeatherIconView
+                                style={styles.forecastItemLong}
+                                key={forecast.EpochDateTime}
+                                forecast={forecast}
+                            />
+                        )
+                    })}
+                </GridRowSplit>
+            </View>
         )
     }
 
-    return <></>
+    return <Text>invalid weather</Text>
 }
 
-const Weather = React.memo(() => {
-    const weatherResponse = useWeatherResponse()
-    return weatherResponse({
-        error: ({}) => <></>,
-        pending: () => <></>,
-        success: weatherForecast => (
-            <WeatherWithForecast
+const GET_WEATHER_DATA = gql`
+    query getWeatherData($locationPermissionStatus: String!) {
+        weatherForecast(locationPermissionStatus: $locationPermissionStatus)
+            @client
+    }
+`
+
+type Props = { locationPermissionStatus: string } & NavigationInjectedProps
+
+/**
+ * We do a local query instead of doing it higher up in the tree so that we
+ * try fetch forecasts only if the weather widget is indeed shown.
+ */
+export const Weather = withNavigation(
+    React.memo(({ locationPermissionStatus, navigation }: Props) => {
+        const weatherData = useQuery(GET_WEATHER_DATA, {
+            variables: { locationPermissionStatus },
+        })
+        if (weatherData.error) console.error(weatherData.error)
```

Comment:
`throw` instead of just `console.error` (will get caught at closest error boundary)